File: src/types/audit.ts
export type AuditActionType = 
  | 'AI_RECOMMENDATION' 
  | 'FIELD_VERIFICATION_SUBMITTED' 
  | 'DATA_CONFLICT_DETECTED' 
  | 'HUMAN_DECISION_OVERRIDE' 
  | 'ROAD_STATUS_UPDATED' 
  | 'RECOVERY_SANCTIONED' 
  | 'SETTLEMENT_APPROVED' 
  | 'EMERGENCY_DISPATCHED';

export type ConflictResolutionChoice = 
  | 'ACCEPT_FIELD' 
  | 'ACCEPT_AI' 
  | 'MODIFY';

export type ConflictStatus = 
  | 'CONFLICT_DETECTED' 
  | 'UNDER_HUMAN_REVIEW' 
  | 'RESOLVED_ACCEPTED_FIELD' 
  | 'RESOLVED_ACCEPTED_AI' 
  | 'RESOLVED_MODIFIED';

export interface AuditActor {
  id: string;
  name: string;
  role: string;
  designation: string;
  badgeNumber: string;
}

export interface AuditLogEntry {
  id: string;
  timestamp: string;
  actor: AuditActor;
  actionType: AuditActionType;
  title: string;
  targetEntityId: string;
  targetEntityType: 'SETTLEMENT' | 'ROAD' | 'SHELTER' | 'INCIDENT' | 'RECOVERY_PROJECT' | 'DATA_FEED';
  targetEntityName: string;
  previousValue: string;
  newValue: string;
  officialReason: string;
  statutoryBasis?: string; // e.g. "Section 30(2)(v) Disaster Management Act 2005"
  confidenceBefore?: number;
  confidenceAfter?: number;
  evidenceReference?: string;
}

export interface DataConflictItem {
  id: string;
  parameterName: string;
  locationName: string;
  settlementId: string;
  settlementName: string;
  
  // AI Sensor Observation
  aiValue: string;
  aiConfidence: number; // e.g. 82%
  aiSource: string; // e.g. "Satellite Synthetic Aperture Radar (SAR) Telemetry"
  aiTimestamp: string;
  
  // Field Officer Ground Truth
  fieldValue: string;
  fieldConfidence: number; // e.g. 96%
  fieldOfficerName: string;
  fieldOfficerBadge: string;
  fieldTimestamp: string;
  fieldEvidenceNotes: string;
  fieldGeotaggedPhotoUrl?: string;
  
  // Discrepancy details
  conflictSeverity: 'CRITICAL' | 'HIGH' | 'MODERATE';
  discrepancySummary: string;
  impactedWorkflows: string[]; // e.g. ["Safe Route Recalculation", "Evacuation Priority", "Citizen Notification"]
  status: ConflictStatus;
  
  // Human Resolution
  resolvedBy?: AuditActor;
  resolvedAt?: string;
  resolutionChoice?: ConflictResolutionChoice;
  officerDecisionReason?: string;
}

export interface AdministrativeReviewStep {
  id: string;
  stageName: string;
  status: 'COMPLETED' | 'IN_PROGRESS' | 'PENDING' | 'REJECTED';
  officerName?: string;
  officerDesignation?: string;
  timestamp?: string;
  notes?: string;
  legalSignOffRequired: boolean;
}

File: backend/app/models/audit.py
from sqlalchemy import Column, String, Integer, Float, Boolean, JSON, Text
from app.core.database import Base


class AuditLogModel(Base):
    __tablename__ = "audit_logs"

    id = Column(String(64), primary_key=True, index=True)
    timestamp = Column(String(64), nullable=False)
    actor_name = Column(String(128), nullable=False)
    actor_role = Column(String(128), nullable=False)
    action_type = Column(String(64), nullable=False)
    title = Column(String(255), nullable=False)
    target_entity_id = Column(String(64), nullable=False)
    target_entity_type = Column(String(64), nullable=False)
    target_entity_name = Column(String(128), nullable=False)
    previous_value = Column(String(255), nullable=True)
    new_value = Column(String(255), nullable=True)
    official_reason = Column(Text, nullable=True)
    statutory_basis = Column(String(255), default="Section 30(2)(v) Disaster Management Act 2005")
    confidence_before = Column(Integer, nullable=True)
    confidence_after = Column(Integer, nullable=True)
    evidence_reference = Column(String(128), nullable=True)


class DataConflictModel(Base):
    __tablename__ = "data_conflicts"

    id = Column(String(64), primary_key=True, index=True)
    parameter_name = Column(String(128), nullable=False)
    location_name = Column(String(128), nullable=False)
    settlement_id = Column(String(64), nullable=False, index=True)
    settlement_name = Column(String(128), nullable=False)

    ai_value = Column(String(128), nullable=False)
    ai_confidence = Column(Integer, default=80)
    ai_source = Column(String(128), default="Satellite Synthetic Aperture Radar (SAR)")

    field_value = Column(String(128), nullable=False)
    field_confidence = Column(Integer, default=95)
    field_officer_name = Column(String(128), default="Revenue Inspector")

    discrepancy_reason = Column(Text, nullable=True)
    impact_level = Column(String(32), default="HIGH")
    status = Column(String(64), default="CONFLICT_DETECTED")
    resolved_choice = Column(String(64), nullable=True)
    resolution_rationale = Column(Text, nullable=True)
